### PR TITLE
Rolling PR for flight-job

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**

### CODE REVIEW

File: builders/flight-job/config/software/flight-job.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  ).instance_exec { self.binding }
+
+  # Renders the reference into the project directory
+  # NOTE: I believe this caches the rendered file TBC
+  rendered_path = 'etc/config.reference.rendered'
+  block do
+    template = File.read(File.join(project_dir, 'etc/config.reference')).gsub(/^#<%/, '<%')
+    reference = ERB.new(template, nil, '-').result(context)
+    File.write(File.join(project_dir, rendered_path), reference)
+  end
```
Comment (benarmston) [inline builders/flight-job/config/software/flight-job.rb:70]: @WilliamMcCumstie IIRC `project_dir` is shared between the different vagrant machines (centos7, centos8, ubuntu*).  If I'm correct, this introduces a race condition when there are multiple concurrent builds on the different distros.

I know that `build_dir` is unique to the vagrant machine, so changing this to write the temporary files into `build_dir` and then copy to `install_dir` would fix the race condition.

Can you either check that `project_dir` is not shared across the vagrant machines, or switch to using `build_dir`.

File: builders/flight-job/opt/flight/usr/share/job/templates/mpi-nodes.sh [hunk 2]
```diff
@@ -0,0 +1,193 @@
+#!/bin/bash -l
+#@      cw_TEMPLATE[name]="MPI multiple node (Slurm)"
+#@      cw_TEMPLATE[desc]="Submit a single job that spans multiple nodes where you want exclusive use of each node allocated."
+#@ cw_TEMPLATE[copyright]="Copyright (C) 2017 Alces Software Ltd."
```
Comment (benarmston) [inline builders/flight-job/opt/flight/usr/share/job/templates/mpi-nodes.sh:4]: The copyright needs to change to `Alces Flight Ltd.`  Here, a couple of lines below and in all other templates.  We should also update the year to `2017-present`.

**Review by benarmston — CHANGES_REQUESTED** [review]
@WilliamMcCumstie I've a couple of comments on this PR that need addressing before its ready to merge.

File: builders/flight-job/config/software/flight-job.rb (hunk 1 above)
Comment (WilliamMcCumstie) [reply builders/flight-job/config/software/flight-job.rb:70]: AFAICS `project_dir` and `build_dir` are stored in essentially the same locations:
```
[vagrant@openflight-builder-builder flight-job]$ ls /home/vagrant/flight-job/local/build/flight-job/
[vagrant@openflight-builder-builder flight-job]$ ls /home/vagrant/flight-job/local/src/flight-job/etc/
config.reference  config.reference.rendered
[vagrant@openflight-builder-builder flight-job]$ ls /home/vagrant/flight-job/local/build/flight-job/
```

IIRC the `vagrant` flotilla of `vms` share their `/vagrant` and `/code` directories. This means both the `project_dir` and `build_dir` are safe in this situation. :dragon: This behaviour depends on the `omnibus.rb` configuration and can not be guaranteed in general.

The `project_dir` is nice as it works out of the box with the `copy` omnibus directive.

File: builders/flight-job/opt/flight/usr/share/job/templates/mpi-nodes.sh (hunk 2 above)
Comment (WilliamMcCumstie) [reply builders/flight-job/opt/flight/usr/share/job/templates/mpi-nodes.sh]: Updated in: https://github.com/openflighthpc/openflight-omnibus-builder/pull/45/commits/4bb8e1e95707797e2abf714e0c7372063a5d54de

File: builders/flight-job/config/software/flight-job.rb (hunk 1 above)
Comment (benarmston) [reply builders/flight-job/config/software/flight-job.rb:70]: OK. I stand corrected: `project_dir` is fine.  Thanks for the clarification.

**Pull request merged**